### ISSUE TITLE
curl documentation lacks information about which conditions must be met for curl to save the cookie jar

### ISSUE THREAD

**Issue by SkunkOS** [curl/curl#12529]
### I did this

I used curl with the cookie and cookie jar options.
I made sure to specify absolute path names, having read that curl silently discards the cookie jar if you only specify a relative pathname.

### I expected the following

I expected curl saving the cookies, including the received and changed ones to the cookie jar.
But it didn't save anything.

### curl/libcurl version

> curl -V
curl 8.4.0 (amd64-portbld-freebsd13.2) libcurl/8.4.0 OpenSSL/1.1.1t zlib/1.2.13 libpsl/0.21.2 (+libidn2/2.3.4) libssh2/1.11.0 nghttp2/1.57.0
Release-Date: 2023-10-11
Protocols: dict file ftp ftps gopher gophers http https imap imaps pop3 pop3s rtsp scp sftp smtp smtps telnet tftp
Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets

### operating system

> uname -a
FreeBSD skunk 13.2-RELEASE-p6 FreeBSD 13.2-RELEASE-p6 releng/13.2-08225c103 SKUNKOS amd64

**Comment by bagder** [comment]
> having read that curl silently discards the cookie jar if you only specify a relative pathname.

Where did you read that?

> I expected curl saving the cookies, including the received and changed ones to the cookie jar. But it didn't save anything.

Please show us what you did!

**Comment by SkunkOS** [comment]
> > having read that curl silently discards the cookie jar if you only specify a relative pathname.
> 
> Where did you read that?
When I googled trying to find out why curl didn't save the cookies, I found a site saying this.
I think I already deleted that browser tab. It was something about using a curl library.

> > I expected curl saving the cookies, including the received and changed ones to the cookie jar. But it didn't save anything.
> 
> Please show us what you did!

Basically, just this: my $ret = `curl -s -b $cookiefilepath -c $cookienewfilepath -K $fnam_curlopt`;

As I had other issues with curl already before, like:

- not downloading the actual files and only saving the headers, if I used the option to log the headers, which would prevent me to keep logs for analysis, and 
- that #http_only comment thing in the curl cookie file which does not seem very compliant with the Netscape cookie file format and would require me to write curl-specific cookie analysis code,

After posting my issue here, I decided to modify my scraper to use wget instead of doing more experiments with curl, trying to somehow persuade it to write the cookie file predictably.
Now the scraper is working just fine using wget.

So I apologize that I cannot look deeper into these issues anymore.
